lib/rspec/matchers/built_in/start_or_end_with.rb

```diff
@@ -56,10 +52,16 @@ def subsets_comparable?
         end
       end
 
+      # For RSpec 3.1, the base class was named `StartAndEndWith`. For SemVer reasons,
+      # we still provide this constant until 4.0.
+      # @deprecated Use StartOrEndWith instead.
+      # @private
+      StartAndEndWith = StartOrEndWith
```

Comment:
When we were building RSpec 3 we would define `const_missing` and define the lookup there to issue a deprecation, is it worth doing that here too?